### PR TITLE
Jax fix

### PR DESCRIPTION
**Pull request by sebasrb09**
Made the following fixes:

- Add abstract loss class
- Create an instance for the CrossEntropyLoss
- Modify small and test models, such that they all can use the same loss
- Modify the tests accordingly, to use the tests
- Fix the examples to also use the Loss
- Add the resizer as an argument for the model_evaluation method
- Add the resizer as static argument

### CODE REVIEW

**Comment by sebasrb09** [comment]
The only ones that create a conflict is the notebooks, since they a re run. It is safe to merge.

File: tests/test_end_to_end_simple.py [hunk 1]
```diff
@@ -73,12 +84,12 @@ def test_simple_end_to_end_non_DP():
...
-            logits = state.apply_fn(resized_X, params=params)
-            one_hot = jax.nn.one_hot(y, num_classes=num_classes)
-            loss = optax.softmax_cross_entropy(logits=logits, labels=one_hot).flatten()
-            return jnp.sum(loss)
+        # def loss_fn(params, X, y):
+        #     resized_X = X
+        #     logits = state.apply_fn(resized_X, params=params)
+        #     one_hot = jax.nn.one_hot(y, num_classes=num_classes)
+        #     loss = optax.softmax_cross_entropy(logits=logits, labels=one_hot).flatten()
+        #     return jnp.sum(loss)
```
Comment (Solosneros) [inline tests/test_end_to_end_simple.py:92]: just remove unused stuff

File: src/jax_mask_efficient.py [hunk 2]
```diff
@@ -173,13 +175,100 @@ def setup_physical_batches_distributed(
...
+        """
+        pass
+
+class CrossEntropyLoss(LossFunction):
+
+    def __init___(
+            self,
+            state,
+            num_classes,
+            resizer
```
Comment (Solosneros) [inline src/jax_mask_efficient.py:232]: could we actually name it `resizer_fn` everywhere?

File: src/jax_mask_efficient.py [hunk 3]
```diff
@@ -173,13 +175,100 @@ def setup_physical_batches_distributed(
...
+
+
 @partial(jax.jit, static_argnums=(3,))
 def compute_per_example_gradients_physical_batch(
     state: train_state.TrainState,
     batch_X: jax.typing.ArrayLike,
     batch_y: jax.typing.ArrayLike,
-    num_classes: int,
-    resizer=None,
+    loss_fn: Callable
```
Comment (Solosneros) [inline src/jax_mask_efficient.py:271]: What do you think about writing LossFunction here instead?

Comment (Solosneros) [reply src/jax_mask_efficient.py:271]: I mean instead of Callable

**Review by Solosneros — CHANGES_REQUESTED** [review]
Looks good, I think I only have minor feedback (see lines). Why exactly are you changing the return of the CNN to tuple? Did it crash before?
